### PR TITLE
Add a param-free classification model 

### PR DESCRIPTION
**Pull request by c4n**
### What does this changes

Add a param-free classification model from [“Low-Resource” Text Classification: A Parameter-Free Classification Method with Compressors](https://aclanthology.org/2023.findings-acl.426) (Jiang et al., Findings 2023)

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/PyThaiNLP/pythainlp/blob/dev/CONTRIBUTING.md) to this repository.

- [x] Passed code styles and structures
- [] Passed code linting checks and unit test

### CODE REVIEW

File: tests/test_cls.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+          ("บริการแย่มากก เป็นหมอได้ไง😤", "Negative"),
+          ("ขับรถแย่มาก", "Negative"),
+          ("ดีนะครับ", "Positive"),
+          ("ลองแล้วรสนี้อร่อย... ชอบๆ", "Positive"),
+          ("ฉันรู้สึกโกรธ เวลามือถือแบตหมด", "Negative"),
+          ("เธอภูมิใจที่ได้ทำสิ่งดี ๆ และดีใจกับเด็ก ๆ", "Positive"),
+          ("นี่เป็นบทความหนึ่ง", "Neutral")
+        ]
+        model = GzipModel(training_data)
+        self.assertIsNotNone(model.predict("รู้สึกดีจัง", k=3))
```
Comment (p16i) [inline tests/test_cls.py]: would it better to modify this assert to also verify the correctness of the prediction?

Comment (p16i) [reply tests/test_cls.py]: also, I get the result from the test:

image

Comment (c4n) [reply tests/test_cls.py]: I don't think we can expect much from this few-shot classifier.   May be sentiment analysis is not the best application to test  either.

Comment (c4n) [reply tests/test_cls.py]: News classification seems to work well  (at least on the paper)

Comment (wannaphong) [reply tests/test_cls.py]: OK. I updated the test.

Comment (p16i) [reply tests/test_cls.py]: > I don't think we can expect much from this few-shot classifier. May be sentiment analysis is not the best application to test either.

I generally agree, but I just wanted to make the test suite slightly less confusing.

**Review by wannaphong — APPROVED** [review]

**Review by p16i — APPROVED** [review]
The implementation looks good to me.

**Comment by p16i** [comment]
@wannaphong  do we have to fix the MacOS error?

**Comment by wannaphong** [comment]
> @wannaphong do we have to fix the MacOS error?

I will fix the MacOS error after this pull request. The error is come from tensorflow (again).

**Pull request merged**